micropython/umqtt.simple/umqtt/simple.py

```diff
@@ -67,15 +65,13 @@ def connect(self, clean_session=True):
         addr = socket.getaddrinfo(self.server, self.port)[0][-1]
         self.sock.connect(addr)
         if self.ssl:
-            import ussl
-
-            self.sock = ussl.wrap_socket(self.sock, **self.ssl_params)
+            self.sock = self.ssl.wrap_socket(self.sock, server_hostname=self.server)
         premsg = bytearray(b"\x10\0\0\0\0\0")
         msg = bytearray(b"\x04MQTT\x04\x02\0\0")
 
         sz = 10 + 2 + len(self.client_id)
         msg[6] = clean_session << 1
-        if self.user is not None:
+        if self.user:
```

Comment:
This should not change.  The existing code supports passing in an empty string for the `user`.

Reply:
Initially, I had `is not None` for all checks, as that was the behavior of the C-code. Doing `if cadata` changes that behavior in the way, that passing `cadata = ""` will now not produce an error anymore, but will configure the field empty.

I've looked around a bit for MQTT implementations that expect an empty username (instead of the field missing), and I couldn't find any. Do you know of any implementation that expects passing an empty string for `user`?

So I just cleaned up more code, but I can undo that again.

Reply:
> I've looked around a bit for MQTT implementations that expect an empty username (instead of the field missing), and I couldn't find any.

OK, then let's keep your change.

For reference the commit that added this line is fcec7f7c1d3485474596a0db38a83e7b9ce5870b